tests/sanity/tests/chat/chat.spec.ts

```diff
@@ -107,49 +105,35 @@ test.describe('channel tests', () => {
     await channelPage.checkIfChannelDefaultExist(true, data.channelName)
     await channelPage.sendMessage('Test message')
     await channelPage.checkMessageExist('Test message', true, 'Test message')
-    await leftSideMenuPage.openProfileMenu()
-    await leftSideMenuPage.inviteToWorkspace()
-    await leftSideMenuPage.getInviteLink()
 
-    const linkText = await page.locator('.antiPopup .link').textContent()
-    using _page = await getSecondPage(browser)
-    const page2 = _page.page
+    const linkText = await getInviteLink(page)
+    await api.createAccount(newUser2.email, newUser2.password, newUser2.firstName, newUser2.lastName)
+    const page2 = await getSecondPageByInvite(browser, linkText, newUser2)
 
     const leftSideMenuPageSecond = new LeftSideMenuPage(page2)
     const channelPageSecond = new ChannelPage(page2)
-    await api.createAccount(newUser2.email, newUser2.password, newUser2.firstName, newUser2.lastName)
-    await page2.goto(linkText ?? '')
-    const joinPage = new SignInJoinPage(page2)
-    await joinPage.join(newUser2)
     await leftSideMenuPageSecond.clickChunter()
     await channelPageSecond.checkIfChannelDefaultExist(false, data.channelName)
     await channelPageSecond.clickChannelTab()
     await channelPageSecond.checkIfChannelTableExist(data.channelName, true)
   })
 
-  test('create new private channel and test if the user can exchange the messages', async ({ browser, page }) => {
+  test('Create new private channel and test if the user can exchange the messages', async ({ browser, page }) => {
     await leftSideMenuPage.clickChunter()
     await chunterPage.clickChannelBrowser()
     await chunterPage.clickNewChannelHeader()
     await chunterPage.createPrivateChannel(data.channelName, false)
     await channelPage.checkIfChannelDefaultExist(true, data.channelName)
     await channelPage.sendMessage('Test message')
     await channelPage.checkMessageExist('Test message', true, 'Test message')
-    await leftSideMenuPage.openProfileMenu()
-    await leftSideMenuPage.inviteToWorkspace()
-    await leftSideMenuPage.getInviteLink()
 
-    const linkText = await page.locator('.antiPopup .link').textContent()
-    using _page = await getSecondPage(browser)
-    const page2 = _page.page
+    const linkText = await getInviteLink(page)
+    await api.createAccount(newUser2.email, newUser2.password, newUser2.firstName, newUser2.lastName)
+    const page2 = await getSecondPageByInvite(browser, linkText, newUser2)
```

Comment:
page2 should be closed, using allow to close it. Please adapt approach

Reply:
I'm working on it right now, will change getSecondPageByInvite like getSecondPage

Reply:
Updated getSecondPageByInvite method. It works like getSecondPage now. Honestly, I think we can find mode elegant solution instead using _page2 in each tests case later 😀